### PR TITLE
Fix build errors with legacy compiler with nojit

### PR DESCRIPTION
**Pull request by Cellule**
I accidentally built with nojit and legacy at the same time and it found some interesting oversight

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/2233)

### CODE REVIEW

**Comment by Cellule** [comment]
I don't think we got any coverage for this scenario and I don't know if we care about it, but I figured I might as well fix them since I stumbled upon it

**Comment by Cellule** [comment]
@MikeHolman Can you take a look at the missing constructors in WebAssembly new objects?

File: lib/Runtime/Language/InterpreterStackFrame.cpp [hunk 1]
```diff
@@ -1163,8 +1163,6 @@ namespace Js
             !(this->executeFunction->GetHasTry() && (PHASE_OFF((Js::JITLoopBodyInTryCatchPhase), this->executeFunction) || this->executeFunction->GetHasFinally())) &&
             (this->executeFunction->ForceJITLoopBody() || this->executeFunction->IsJitLoopBodyPhaseEnabled()) &&
             !this->executeFunction->IsInDebugMode();
-#else
-        const bool doJITLoopBody = false;
```
Comment (dilijev) [inline lib/Runtime/Language/InterpreterStackFrame.cpp]: What happened here?

Comment (Cellule) [inline lib/Runtime/Language/InterpreterStackFrame.cpp]: This variable is unused in nojit

File: lib/Runtime/Library/WebAssemblyMemory.h [hunk 2]
```diff
@@ -9,6 +9,9 @@ namespace Js
 {
     class WebAssemblyMemory : public DynamicObject
     {
+    protected:
+        DEFINE_VTABLE_CTOR( WebAssemblyMemory, DynamicObject );
+        DEFINE_MARSHAL_OBJECT_TO_SCRIPT_CONTEXT( WebAssemblyMemory );
```
Comment (MikeHolman) [inline lib/Runtime/Library/WebAssemblyMemory.h]: Is this related to legacy no jit? I talked with @boingoing and I think there is more to do here.

Comment (Cellule) [inline lib/Runtime/Library/WebAssemblyMemory.h]: When I tried to build nojit with vs2013 it complained that WebAssemblyMemory/Table didn't have a constructor.
I added this by looking at other javascript objects, but I don't know if it needs something else

Comment (MikeHolman) [inline lib/Runtime/Library/WebAssemblyMemory.h]: ok. I think this is fine to add, but we may still have more to do for marshalling to really work. We can talk to Curtis in January for that though.

**Comment by dilijev** [comment]
Okay, sounds like this is good to merge?

**Comment by Cellule** [comment]
Yes I just didn't have time before the weekend, thanks for reminding me
